lib/status.js

```diff
@@ -1,78 +1,112 @@
+    break;
+  case 'getBestBlockHash':
+    this.getBestBlockHash(function(err, result) {
+      if (err) {
+        return common.handleErrors(err, res);
+      }
+      res.jsonp(result);
+    });
+    break;
+  case 'getInfo':
+  default:
+    this.getInfo(function(err, result) {
+      if (err) {
+        return common.handleErrors(err, res);
+      }
+      res.jsonp({
+        info: result
+      });
+    });
   }
 };
 
-StatusController.prototype.getInfo = function() {
-  var info = this.node.services.bitcoind.getInfo();
-  return {
-    info: info
-  };
+StatusController.prototype.getInfo = function(callback) {
+  this.node.services.bitcoind.getInfo(callback);
 };
 
 StatusController.prototype.getLastBlockHash = function() {
-  var hash = this.node.services.db.tip.hash;
+  var hash = this.node.services.bitcoind.tiphash;
   return {
     syncTipHash: hash,
     lastblockhash: hash
   };
 };
 
-StatusController.prototype.getBestBlockHash = function() {
-  var hash = this.node.services.bitcoind.getBestBlockHash();
-  return {
-    bestblockhash: hash
-  };
+StatusController.prototype.getBestBlockHash = function(callback) {
+  this.node.services.bitcoind.getBestBlockHash(function(err, hash) {
+    if (err) {
+      return callback(err);
+    }
+    callback(null, {
+      bestblockhash: hash
+    });
+  });
 };
 
-StatusController.prototype.getDifficulty = function() {
-  var info = this.node.services.bitcoind.getInfo();
-  return {
-    difficulty: info.difficulty
-  };
+StatusController.prototype.getDifficulty = function(callback) {
+  this.node.services.bitcoind.getInfo(function(err, info) {
+    if (err) {
+      return callback(err);
+    }
+    callback(null, {
+      difficulty: info.difficulty
+    });
+  });
 };
 
 StatusController.prototype.sync = function(req, res) {
+  var self = this;
   var status = 'syncing';
-  if(this.node.services.bitcoind.isSynced() && this.node.services.db.tip.__height === this.node.services.bitcoind.height) {
-    status = 'finished';
-  }
 
-  // Not exactly the total blockchain height,
-  // but we will reach 100% when our db and bitcoind are both fully synced
-  var totalHeight = this.node.services.bitcoind.height / (this.node.services.bitcoind.syncPercentage() / 100);
-
-  var info = {
-    status: status,
-    blockChainHeight: this.node.services.bitcoind.height,
-    syncPercentage: Math.round(this.node.services.db.tip.__height / totalHeight * 100),
-    height: this.node.services.db.tip.__height,
-    error: null,
-    type: 'bitcore node'
-  };
+  this.node.services.bitcoind.isSynced(function(err, synced) {
+    if (err) {
+      return common.handleErrors(err, res);
+    }
+    if (synced) {
+      status = 'finished';
+    }
+
+    self.node.services.bitcoind.syncPercentage(function(err, percentage) {
+      if (err) {
+        return common.handleErrors(err, res);
+      }
+      var info = {
+        status: status,
+        blockChainHeight: self.node.services.bitcoind.height,
```

Comment:
This is for backwards compatibility and carry-over from the info method at: https://github.com/bitpay/insight-api/blob/prod2/lib/HistoricSync.js where these could have been different values.